azurerm/internal/services/apimanagement/tests/api_management_resource_test.go

```diff
@@ -240,6 +242,347 @@ func testCheckAzureRMApiManagementExists(resourceName string) resource.TestCheck
 	}
 }
 
+func TestAccAzureRMApiManagement_identityUserAssigned(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_api_management", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMApiManagementDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMApiManagement_identityUserAssigned(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMApiManagementExists(data.ResourceName),
+					resource.TestCheckResourceAttr(data.ResourceName, "identity.type", "UserAssigned"),
+					resource.TestCheckResourceAttr(data.ResourceName, "identity.identity_ids.#", "1"),
+					resource.TestCheckNoResourceAttr(data.ResourceName, "identity.principal_id"),
+					resource.TestCheckNoResourceAttr(data.ResourceName, "identity.tenant_id"),
```

Comment:
I am so sorry I did not notice this earlier - now in the acc tests, the import step will do the checks for you to ensure the state after an apply is exactly the same as the config, therefore we could omit these manual checks in resource tests.

But, in the data source tests, since we cannot add the import step (which will lead to errors), we will still have to do the checks manually.

Reply:
Thanks! Should I still keep these lines, or are they also not needed?
```
testCheckAzureRMApiManagementExists(data.ResourceName),
```